lib/TPP/Conversion/ConvertLinalgToXsmm/ConvertLinalgToXsmm.cpp

```diff
@@ -575,9 +573,8 @@ static FailureOr<BrgemmInfo> checkAccess(linalg::LinalgOp linalgOp, unsigned m,
   }
 
   auto loops = linalgOp.computeStaticLoopSizes();
-  int64_t batchVal = (batch != std::numeric_limits<unsigned>::max())
-                         ? loops[batch]
-                         : std::numeric_limits<int64_t>::max();
+  int64_t batchVal =
+      (batchPos != std::numeric_limits<unsigned>::max()) ? loops[batchPos] : 0;
```

Comment:
Why would `batchPos` be `MAX_INT` here? You can easily replace the existing callers to change to zero and not have to worry about int `max` any more anywhere.

Reply:
We cannot use 0 for batchPos because it represent the position of the batch dimension in the domain of the generic.

Reply:
The thing is, you're encoding a "not-a-number" value type with an "inf" value and that is confusing. Can we perhaps use `std::optional` here?

Reply:
We could, but this would mean treating the batch in a special way different from the other dimensions like m, n, and k and encoding the special case in methods like: `getPosInCodomain.`

Reply:
Fair, let's postpone this discussion.

Reply:
Let me merge this and create a PR on this, sounds good?

Reply:
if we don't like the changes we can discard the PR.

Reply:
Sounds good, thanks!